ambari-server/src/main/java/org/apache/ambari/server/stack/StackDirectory.java

```diff
@@ -478,4 +526,52 @@ private void parseRoleCommandOrder() {
       LOG.error(String.format("Can not read role command order info %s", rcoFilePath), e);
     }
   }
+
+  /**
+   * Loads any classes found in the {@link #libraryDir} and creates a
+   * {@link URLClassLoader} for them.
+   */
+  private void parseLibraryClassLoader() {
+    if (null == libraryDir) {
+      return;
+    }
+
+    Path libraryPath = Paths.get(libraryDir);
+    if( Files.notExists(libraryPath) || !Files.isDirectory(libraryPath) ) {
+      return;
+    }
+
+    try {
+      List<URI> jarUris = Files.list(libraryPath)
+          .filter(file -> file.toString().endsWith(".jar"))
+          .map(Path::toUri)
+          .collect(Collectors.toList());
+
+      List<URL> jarUrls = new ArrayList<>(jarUris.size());
+      for (URI jarUri : jarUris) {
+        try {
+          jarUrls.add(jarUri.toURL());
+        } catch (MalformedURLException malformedURLException) {
+          LOG.error("Unable to load the stack library {}", jarUri, malformedURLException);
+        }
+      }
+
+      URL[] jarUrlArray = new URL[jarUris.size()];
+      libraryClassLoader = new URLClassLoader(jarUrls.toArray(jarUrlArray),
+          ClassUtils.getDefaultClassLoader());
+
+      try {
+        @SuppressWarnings("unchecked")
+        Class<UpgradeCheck> clazz = (Class<UpgradeCheck>) libraryClassLoader.loadClass(
+            "testbed.FooCheck");
```

Comment:
Can you please explain `testbed.FooCheck`?

Reply:
Hey, nice catch! This was leftover from me testing custom provided 3rd party libraries.